### PR TITLE
[PLAT-450] Consolidate all configs

### PR DESCRIPTION
**Pull request by DidacSF**
## Description

Check for details. In short:

* Consolidated `MintFee`, `MintAvailable` and `MintCooldown` into a single struct: `SeasonConfigs`

## Type of changes

- [ ] `build`: Changes that affect the build system or external dependencies (eg, Cargo, Docker)
- [ ] `ci`: Changes to CI configuration
- [ ] `docs`: Changes to documentation only
- [ ] `feat`: Changes to add a new feature
- [ ] `fix`: Changes to fix a bug
- [x] `refactor`: Changes that do not alter functionality
- [ ] `style`: Changes to format the code
- [ ] `test`: Changes to add missing tests or correct existing tests

## Checklist

- [x] ~Tests for the changes have been added~
- [x] Necessary documentation is added (if appropriate)
- [x] Formatted with `cargo fmt --all`
- [x] Linted with `cargo clippy --all-features --all-targets`
- [x] Tested with `cargo test --workspace --all-features --all-targets`

### CODE REVIEW

File: pallets/ajuna-awesome-avatars/src/lib.rs [hunk 1]
```diff
@@ -43,6 +43,25 @@ pub mod pallet {
 	};
 	use sp_std::vec::Vec;
 
+	#[derive(Debug, Encode, Decode, MaxEncodedLen, TypeInfo)]
+	pub struct SeasonConfigs<Balance, BlockNumber> {
+		pub mint_available: bool,
+		pub mint_fees: Balance,
```
Comment (cowboy-bebug) [inline pallets/ajuna-awesome-avatars/src/lib.rs:49]: Let's try getting https://github.com/ajuna-network/Ajuna/pull/74 merged in first so this becomes:
```suggestion
		pub mint_fees: MintFees<Balance>,
```

File: pallets/ajuna-awesome-avatars/src/lib.rs [hunk 2]
```diff
@@ -43,6 +43,25 @@ pub mod pallet {
 	};
 	use sp_std::vec::Vec;
 
+	#[derive(Debug, Encode, Decode, MaxEncodedLen, TypeInfo)]
+	pub struct SeasonConfigs<Balance, BlockNumber> {
```
Comment (cowboy-bebug) [inline pallets/ajuna-awesome-avatars/src/lib.rs:47]: `SeasonConfigs` sounds good :+1:

File: pallets/ajuna-awesome-avatars/src/lib.rs [hunk 3]
```diff
@@ -102,27 +121,9 @@ pub mod pallet {
...
-	pub fn DefaultMintCooldown<T: Config>() -> T::BlockNumber {
-		5_u8.into()
-	}
-
-	#[pallet::storage]
-	#[pallet::getter(fn mint_cooldown)]
-	pub type MintCooldown<T: Config> =
-		StorageValue<_, T::BlockNumber, ValueQuery, DefaultMintCooldown<T>>;
+	#[pallet::getter(fn configs)]
+	pub type Configs<T: Config> =
```
Comment (cowboy-bebug) [inline pallets/ajuna-awesome-avatars/src/lib.rs:125]: Let's rename for consistency if we can:
```suggestion
	#[pallet::getter(fn season_configs)]
	pub type SeasonConfigs<T: Config> =
```

File: pallets/ajuna-awesome-avatars/src/lib.rs [hunk 4]
```diff
@@ -43,6 +43,25 @@ pub mod pallet {
...
+		pub mint_available: bool,
+		pub mint_fees: Balance,
+		pub mint_cooldown: BlockNumber,
+	}
+
+	pub type SeasonConfigsOf<T> =
+		SeasonConfigs<BalanceOf<T>, <T as frame_system::Config>::BlockNumber>;
+
+	#[pallet::type_value]
+	pub fn DefaultConfigs<T: Config>() -> SeasonConfigsOf<T> {
```
Comment (cowboy-bebug) [inline pallets/ajuna-awesome-avatars/src/lib.rs:57]: Just for consistency:
```suggestion
	pub fn DefaultSeasonConfigs<T: Config>() -> SeasonConfigsOf<T> {
```

**Review by cowboy-bebug — COMMENTED** [review]
LGTM! It's very tidy :medal_sports:
Let's rebase once https://github.com/ajuna-network/Ajuna/pull/74 is merged.

**Comment by cowboy-bebug** [comment]
Actually, let's rename everything to `GeneralConfig` because these values are not specific to seasons.
(season specific config values are included in `Season` struct itself)

**Comment by DidacSF** [comment]
> Actually, let's rename everything to `GeneralConfig` or `GlobalConfig` (or similar) because these values are not specific to seasons. (season specific config values are included in `Season` struct itself)

I've renamed to struct to `GlobalConfig` and the storage type to `GlobalConfigs`

File: pallets/ajuna-awesome-avatars/src/lib.rs (hunk 1 above)
Comment (DidacSF) [reply pallets/ajuna-awesome-avatars/src/lib.rs]: Finished rebase and code adaptations!

**Review by cowboy-bebug — APPROVED** [review]

**Pull request merged**